packages/graphql/mixin.core.js

```diff
@@ -1,55 +1,69 @@
 const { Mixin } = require('hops-mixin');
-const strip = require('strip-indent');
+const renderMiddleware = require('@untool/webpack/lib/middleware/render');
+const { StatsFilePlugin } = require('@untool/webpack/lib/plugins/stats');
 
 class GraphQLMixin extends Mixin {
-  registerCommands(yargs) {
-    return yargs.command('graphql', 'Execute GraphQL specific tasks', yargs =>
-      yargs
-        .usage('Usage: hops graphql <command>')
-        .command({
-          command: 'introspect',
-          describe: 'Fetches GraphQL schema information for introspection',
-          builder: {
-            header: {
-              alias: 'H',
-              type: 'array',
-              default: [],
-              describe: strip(`
-                Additional HTTP headers to be used when executing the schema
-                introspection on the remote server. Specify this multiple
-                times to add more headers.\nFormat: "Header: Value"
-              `),
-            },
-          },
-          handler: argv => {
-            require('./lib/fragments')({
-              graphqlUri: this.config.graphqlUri,
-              schemaFile: this.config.graphqlSchemaFile,
-              fragmentsFile: this.config.fragmentsFile,
-              headers: argv.header,
-            })
-              .then(() => {
-                console.log('Fetched and saved GraphQL fragments');
-              })
-              .catch(err => {
-                console.error('Could not fetch GraphQL fragments:');
-                console.trace(err);
-              });
-          },
-        })
-        .help('help')
-        .alias('h', 'help')
-        .demandCommand()
+  configureServer(rootApp, middleware) {
+    if (!this.config.graphqlMocks || !this.config.enableGraphqlMockServer) {
```

Comment:
What do you propose we do, when we render static pages? In my opinion the Apollo Server should start for static rendering, so that the rendered pages contain all the data, right?
Controlling if/how the pages should be rendered can be done via `shouldPrefetchOnServer`.

Reply:
IIRC `static` is set when running `hops build`, when running `yarn serve` the mode is `serve`.

Reply:
No, that's not entirely correct. The mode argument can either be `develop`, `serve` or `static`.
- `develop` means that the server has been started in development mode `hops develop` or `hops start` and in that case we want to start the apollo mock server.
- `serve` means that the server has been started in serving mode - started via `hops serve` or `hops start -p`.
- `static` is only there for static rendering and will only occur when a static build is started: `hops build -s` or `hops start -ps`.

So, I'd say the configuration that makes the most sense would be to only run the apollo server is when `mode === develop || mode === static` for now. And for `mode === serve` we should try to find another solution. For example we could use `configureBuild` to create a production ready apollo mock server (similar to how we are building the `server.js` middleware for production builds).